projects/plugins/jetpack/extensions/blocks/contact-info/index.js

```diff
@@ -51,6 +52,50 @@ export const settings = {
 		align: [ 'wide', 'full' ],
 		html: false,
 	},
+	// Transform from classic widget
+	transforms: {
+		from: [
+			{
+				type: 'block',
+				blocks: [ 'core/legacy-widget' ],
+				isMatch: ( { idBase, instance } ) => {
+					if ( ! instance?.raw ) {
+						return false;
+					}
+					return idBase === 'widget_contact_info';
+				},
+				transform: ( { instance } ) => {
+					let innerBlocks = [
+						createBlock( 'core/heading', {
+							content: instance.raw.title,
+						} ),
+						createBlock( 'jetpack/email', {
+							email: instance.raw.email,
+						} ),
+						createBlock( 'jetpack/phone', {
+							phone: instance.raw.phone,
+						} ),
+						createBlock( 'jetpack/address', {
+							address: instance.raw.address,
+						} ),
+					];
+
+					if ( instance.raw.hours ) {
```

Comment:
You might need to make all of the innerBlocks conditional. Otherwise, you end up with blanks blocks if it is empty. This is what it transformed to for me with certain fields blank - https://d.pr/i/8UNAOy

Reply:
I was initially thinking about making all fields conditional, but then decided to leave some of them unconditional. The reasoning here is that if the user is transforming to the Contact Info block, they will see what fields they can fill in (even if they didn't have them filled in the widget previously).

Also, for example, if the widget title is empty, they will see a nice empty header they can fill in. At least this is how I would like it to be (if I were a user). But I get your point!

Reply:
I totally see your point. Generally speaking, I imagine the transform being used for someone who already has a fully built widget and we are just preserving/transferring the settings to the new format. In that case I imagine it could be unnecessary.

It could go either way I guess haha.